src/dfx/src/commands/canister/delete.rs

```diff
@@ -177,28 +177,35 @@ async fn delete_canister(
                 let status =
                     canister::get_canister_status(env, canister_id, &CallSender::SelectedId)
                         .await?;
-                let mut cycles = status.cycles.0.to_u128().unwrap();
-                if cycles > WITHDRAWAL_COST {
-                    cycles -= WITHDRAWAL_COST;
-
-                    if !to_dank {
+                let cycles = status.cycles.0.to_u128().unwrap();
+                let mut attempts = 0_u128;
+                loop {
```

Comment:
The only thing I wondered about was putting the `get_canister_status` call in the loop, since the cycles value will drop, but it's probably not enough to matter.

Reply:
I thought about it too but decided that the extra time the additional update call takes would not be worth the added precision